pkgs/applications/misc/sweethome3d/linux.nix

```diff
@@ -0,0 +1,132 @@
+{
+  lib,
+  stdenv,
+  fetchzip,
+  fetchurl,
+  makeWrapper,
+  makeDesktopItem,
+  jdk,
+  ant,
+  stripJavaArchivesHook,
+  gtk3,
+  gsettings-desktop-schemas,
+  p7zip,
+  autoPatchelfHook,
+  libXxf86vm,
+  libGL,
+
+  pname,
+  version,
+  src,
+  meta,
+}:
+
+let
+
+  # TODO: Should we move this to `lib`? Seems like its would be useful in many cases.
+  extensionOf =
+    filePath: lib.concatStringsSep "." (lib.tail (lib.splitString "." (builtins.baseNameOf filePath)));
+
+  installIcons =
+    iconName: icons:
+    lib.concatStringsSep "\n" (
+      lib.mapAttrsToList (size: iconFile: ''
+        mkdir -p "$out/share/icons/hicolor/${size}/apps"
+        ln -s -T "${iconFile}" "$out/share/icons/hicolor/${size}/apps/${iconName}.${extensionOf iconFile}"
+      '') icons
+    );
+
+  icons = {
+    "32x32" = fetchurl {
+      url = "http://sweethome3d.cvs.sourceforge.net/viewvc/sweethome3d/SweetHome3D/deploy/SweetHome3DIcon32x32.png";
+      sha256 = "1r2fhfg27mx00nfv0qj66rhf719s2g1vhdis7bdc9mqk9x0mb0ir";
+    };
+    "48x48" = fetchurl {
+      url = "http://sweethome3d.cvs.sourceforge.net/viewvc/sweethome3d/SweetHome3D/deploy/SweetHome3DIcon48x48.png";
+      sha256 = "1ap6d75dyqqvx21wddvn8vw2apq3v803vmbxdriwd0dw9rq3zn4g";
+    };
+  };
+
+in
+stdenv.mkDerivation (finalAttrs: {
+  inherit pname version src;
+
+  sweethome3dItem = makeDesktopItem {
+    desktopName = "Sweet Home 3D";
+    name = finalAttrs.pname;
+    icon = finalAttrs.pname;
+    comment = meta.description;
+    exec = meta.mainProgram;
+    genericName = "Computer Aided (Interior) Design";
+    categories = [
+      "Graphics"
+      "2DGraphics"
+      "3DGraphics"
+    ];
+  };
+
+  postPatch = ''
+    addAutoPatchelfSearchPath ${jdk}/lib/openjdk/lib/
+    autoPatchelf lib
+
+    # Nix cannot see the runtime references to the paths we just patched in
+    # once they've been compressed into the .jar. Scan for and remember them
+    # as plain text so they don't get overlooked.
+    find . -name '*.so' | xargs strings | { grep '/nix/store' || :; } >> ./.jar-paths
+  '';
+
+  nativeBuildInputs = [
+    makeWrapper
+    autoPatchelfHook
+    stripJavaArchivesHook
+  ];
+  buildInputs = [
+    ant
+    jdk
+    p7zip
+    gtk3
+    gsettings-desktop-schemas
+    libXxf86vm
+  ];
+
+  # upstream targets Java 7 by default
+  env.ANT_ARGS = "-DappletClassSource=8 -DappletClassTarget=8 -DclassSource=8 -DclassTarget=8";
+
+  buildPhase = ''
+    runHook preBuild
+
+    ant furniture textures help
+    mkdir -p $out/share/{java,applications}
+    mv "build/"*.jar $out/share/java/.
+    ant
+
+    runHook postBuild
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/bin
+    cp install/SweetHome3D-${finalAttrs.version}.jar $out/share/java/.
+
+    ${installIcons finalAttrs.pname icons}
+
+    cp "${finalAttrs.sweethome3dItem}/share/applications/"* $out/share/applications
```

Comment:
```suggestion
```
we have a helper for that https://github.com/NixOS/nixpkgs/blob/e6914ee14cc9ef2221e9ac323c59cab1ae029684/doc/build-helpers/trivial-build-helpers.chapter.md#usage-2-of-makedesktopitem